File: week6/mario-less.py
from cs50 import get_int

# Prompt user for height between 1 and 8
while True:
    h = get_int("Height: ")
    if h > 0 and h < 9:
        break  # exit the loop when conditions are met, continue

# Print the pyramid aligned to the right
for i in range(0, h, 1):  # line - from position 0 to total height, 1 by 1
    for j in range(0, h, 1):  # column - from position 0 to total height, 1 by 1
        if (i + j < h - 1):  # print spaces
            print(" ", end="")  # end: so it doesn't print next line
        else:
            print("#", end="")
    print() # new line

File: week1/mario-less.c
#include <cs50.h>
#include <stdio.h>

// declare function that will appear later on
void print_row(int spaces, int bricks);

int main(void)
{
    // Prompt the user for the pyramid's height
    int n;
    do
    {
        n = get_int("Height: ");
    }
    while (n < 1 || n > 8);

    // Print a pyramid of that height
    for (int i = 0; i < n; i++)
    {
        // Print row of bricks
        int spaces = n - i - 1;
        int bricks = i + 1;
        print_row(spaces, bricks);
    }
}

// Function to print row of the pyramid
void print_row(int spaces, int bricks)
{
    // Print spaces to the left
    for (int j = 0; j < spaces; j++)
    {
        printf(" ");
    }

    // Print bricks
    for (int i = 0; i < bricks; i++)
    {
        printf("#");
    }

    // Print new line
    printf("\n");
}